src/adapter/http/src/platform/router.rs

```diff
@@ -10,19 +10,31 @@
 use utoipa_axum::router::OpenApiRouter;
 use utoipa_axum::routes;
 
+use crate::AuthPolicyLayer;
+
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-pub fn root_router() -> OpenApiRouter {
+pub fn root_router(allow_anonymous: bool) -> OpenApiRouter {
     use crate::platform::handlers;
 
-    OpenApiRouter::new()
+    let mut router = OpenApiRouter::new();
+
+    router = router
+        .routes(routes!(handlers::file_upload_prepare_post_handler))
+        .routes(routes!(handlers::file_upload_post_handler))
+        .routes(routes!(handlers::file_upload_get_handler));
+
+    if !allow_anonymous {
+        router = router.layer(AuthPolicyLayer::new()); // Mutate router without moving
+    }
+
+    router = router
```

Comment:
Maybe make some comments about the endpoints above requiring auth, and the endpoints below not-requiring it